crates/volta-core/src/run/parser.rs

```diff
@@ -126,6 +126,71 @@ impl<'a> CommandArg<'a> {
         }
     }
 
+    /// Parse the given set of arguments to see if they correspond to an intercepted pnpm command
+    pub fn for_pnpm<S>(args: &'a [S]) -> CommandArg<'a>
+    where
+        S: AsRef<OsStr>,
+    {
+        // If VOLTA_UNSAFE_GLOBAL is set, then we always skip any global parsing
+        if env::var_os(UNSAFE_GLOBAL).is_some() {
+            return CommandArg::Standard;
+        }
+
+        if args.is_empty() {
+            return CommandArg::Standard;
+        }
+
+        let args = args
+            .into_iter()
+            .map(AsRef::<OsStr>::as_ref)
+            .collect::<Vec<&OsStr>>();
+
+        let (flags, positionals): (Vec<&OsStr>, Vec<&OsStr>) =
+            args.into_iter().partition(|arg| is_flag(arg));
+
+        if positionals.len() < 2 {
+            return CommandArg::Standard;
+        }
```

Comment:
1. Improved.
2. Improved.
3. I've updated this part of the implementation, please take another review when you are available.

Currently, the interception parser is dead code and will not be used, since all global commands are captured by https://github.com/volta-cli/volta/pull/1273/files#diff-b72f8a187559437e8f61843775ed13a80245c48032061c623123992d89770e44R17-R26 with the _unimplemented note_ early return.